### PR TITLE
Travis CI setup for repo

### LINKED ISSUE #98: Setup Travis CI in the project
Please Note:  **We are preparing to participate in Google Code-in, and have reserved this issue for participants in GCI - but we'd love to have your help with another one! Please check out https://code.publiclab.org to see more**

**What's the feature request? Please describe.**

Travis CI makes developing software efficient and it ensures we deploy clean code that follows good practice and also detects if there are possible bugs or deficits in our code caused by a change or refactor in our project.

**Describe the solution you'd like**

So we should set up Travis CI in this project by creating a file `.travis.yml` in the root directory with proper configuration.

**Additional context**

This service simply means to automate the process by running test cases each time a new commit is pushed.

Thank You 😊

### PR DESCRIPTION
**Pull request by Uzay-G**
Fixes #98

Hey, I managed to configure a working travis-ci build for the repository. It launches a server whenever someone pushes and then tests the changes on the server. 
You can see an example build [here](https://travis-ci.org/Uzay-G/simple-data-grapher/builds/621826272). 
For some reason though I am getting an error on the tests:
```
 1 failing
  1) csv string file upload test
       should select graph type:
     TypeError: Cannot read property 'click' of null
```
I don't think it's because of the Travis environment though so I believe the build works.
What do you think @debck @starkblaze01? Thanks :smile:

### CODE REVIEW

**Comment by Uzay-G** [comment]
I don't understand the error. Did this issue already exist with testing?

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  # Enable user namespace cloning
+  - "sysctl kernel.unprivileged_userns_clone=1"
+  # Launch XVFB
+  - "export DISPLAY=:99.0"
+  - "sh -e /etc/init.d/xvfb start"
+dist: trusty
+sudo: required
+
+script: 
+  - forever start ./node_modules/http-server/bin/http-server -p 8000
```
Comment (chen-robert) [inline .travis.yml]: Do you need to start a server for the travis build? From my understanding, Travis was used to run tests to ensure that the code still works. Maybe try removing this line and see if that fixes the error

**Comment by Uzay-G** [comment]
@debck @starkblaze01 I found the error for the travis build. [This test](https://github.com/publiclab/simple-data-grapher/blob/main/spec/js/ui_tests.js#L96) looks for an element with id 'graph_type2' that doesn't exist on any of the pages. I think something might have changed in the codebase and the test did not get updated.

File: .travis.yml (hunk 1 above)
Comment (Uzay-G) [reply .travis.yml]: No the server needs to be running. I found the issue and am going to fix it now but thanks

**Comment by Uzay-G** [comment]
I need to update the test somehow.

**Comment by chen-robert** [comment]
Oh I see what you mean now @Uzay-G, could you try adding 
```javascript
console.log(graph_type)
```
to see if the element `$("#graph_type2")` exists anywhere? I think it might be dynamically generated

**Comment by Uzay-G** [comment]
Yeah good idea. I just tried and it returned undefined so yeah I think I need to update the test.

**Comment by chen-robert** [comment]
The commit which made this test was `90ad4109769f14778dea0f56594802367c370189`. If you do `git checkout 90ad4109769f14778dea0f56594802367c370189`, you can see what the repository was like when the test was first made, which might help? 

After running 
```
grep -rnw graph_type .
```
I found the following line of code.
```
./src/View.js:307:                radio.id="graph_type"+count;
```

It seems the `graph_type2` id was constructed dynamically here!

**Comment by chen-robert** [comment]
This line of code has been refactored, but seems to correspond to https://github.com/publiclab/simple-data-grapher/blob/main/src/View.js#L371

**Comment by Uzay-G** [comment]
Wait so you think there is an element with id graph_type2

**Comment by chen-robert** [comment]
I think there used to be one, when the test was first made. But then the code got changed which changed the id as well. I provided a link to where the new id might be set, if you want to dig a bit deeper

**Comment by Uzay-G** [comment]
It's quite late where I am so I will remove the deprecated test for now and will try to find a replacement tomorrow.

**Comment by Uzay-G** [comment]
Hey @debck and @IshaGupta18, is this good to merge?
Thanks

**Pull request merged**